src/routes/sessions/[id]/+page.svelte

```diff
@@ -202,6 +200,41 @@
 		session.messages = [...session.messages, message];
 	}
 
+	function resetPrompt() {
+		// Reset the prompt to the last sent message
+		prompt = promptCached;
+		promptCached = '';
+	}
+
+	function abortOllama() {
+		ollamaInstance?.abort();
+		completion = '';
+		// Remove the "incomplete" AI response
+		session.messages = session.messages.slice(0, -1);
+	}
+
+	function showAbortError() {
+		toast.error("Can't stop completion until the first response from Ollama is received");
+	}
+
+	beforeNavigate((navigation) => {
+		if (isWaitingForResponse) {
+			showAbortError();
+			navigation.cancel();
+			return;
+		}
+		if (completion) {
+			const userConfirmed = confirm(
+				'Are you sure you want to leave?\nThe completion in progress will stop'
+			);
+			if (userConfirmed) {
+				abortOllama();
+				return;
+			}
+			navigation.cancel();
+		}
+	});
```

Comment:
And we can simplify this function as:

```suggestion
	beforeNavigate((navigation) => {
		if (!completion) return;
		const userConfirmed = confirm(
			'Are you sure you want to leave?\nThe completion in progress will stop'
		);
		if (!userConfirmed) navigation.cancel();
		stopResponse(navigation);
	});
```

Reply:
I'm afraid this refactored code is not working as it did before. If I click `Run` and then I try to navigate to another page, completion is still `= ''` so the `beforeNavigate` function returns immediately and it does not show the warning toast.

Reply:
Understood, that makes sense.